### PR TITLE
Event eusines house eusine

### LINKED ISSUE #691: Eusine's EusinesHouse Event flag and Eusine's Gramps Text
Eusine's `object_event` should really use his own EVENT flag and not `EVENT_SET_WHEN_FOUGHT_HO_OH`. Or at least it should be renamed to `EVENT_EUSINES_HOUSE_EUSINE`  for consistency. 

[https://github.com/Rangi42/polishedcrystal/blob/bd9eea7f54ba3714b7f2a9feeaada89cb573ae3a/maps/EusinesHouse.asm#L15](https://github.com/Rangi42/polishedcrystal/blob/bd9eea7f54ba3714b7f2a9feeaada89cb573ae3a/maps/EusinesHouse.asm#L15)

Also, it's a bit confusing his order of events..

`EVENT_SET_WHEN_FOUGHT_HO_OH` is initially set here, so EUSINE in EUSINE's House is hidden:

https://github.com/Rangi42/polishedcrystal/blob/eb16651b3d9490a4bea7d6e0043789a288c19d26/data/events/initialize_events.asm#L121

It is then cleared after battling SUICUNE in TinTower1F, thus spawning EUSINE in his house.

[https://github.com/Rangi42/polishedcrystal/blob/66d9860e1f6315a01cd3332ada19a635cc719de1/maps/TinTower1F.asm#L146](https://github.com/Rangi42/polishedcrystal/blob/66d9860e1f6315a01cd3332ada19a635cc719de1/maps/TinTower1F.asm#L146)

The event is set again with the `disappear` command in Eusine's House when Eusine leaves to return to Johto in search of Ho-Oh:

[https://github.com/Rangi42/polishedcrystal/blob/66d9860e1f6315a01cd3332ada19a635cc719de1/maps/EusinesHouse.asm#L78](https://github.com/Rangi42/polishedcrystal/blob/66d9860e1f6315a01cd3332ada19a635cc719de1/maps/EusinesHouse.asm#L78)

The event is set _again_ after battling Ho-Oh in the TinTower roof:

[https://github.com/Rangi42/polishedcrystal/blob/66d9860e1f6315a01cd3332ada19a635cc719de1/maps/TinTowerRoof.asm#L48](https://github.com/Rangi42/polishedcrystal/blob/66d9860e1f6315a01cd3332ada19a635cc719de1/maps/TinTowerRoof.asm#L48)

Result is that EUSINE never returns to kanto and grandfather always states that his grandson is in Johto.

Also Eusine's Grandfather always states the following when EUSINE is away. Even _if_ Eusine is actually looking for Ho-Oh and not Suicune.

https://github.com/Rangi42/polishedcrystal/blob/66d9860e1f6315a01cd3332ada19a635cc719de1/maps/EusinesHouse.asm#L175-L185

### PR DESCRIPTION
**Pull request by vulcandth**
Resolves #691. Also may be related to #208.

Changes:

*   Renames `EVENT_SET_WHEN_FOUGHT_HO_OH` to `EVENT_EUSINES_HOUSE_EUSINE`
*   Adds Additional Eusines House Gramps dialog for when Eusine is away looking for Ho-Oh.

To-do:

*   [ ] Eusine should ideally return to Kanto after the player defeats Ho-Oh, and not remain in Tin-Tower forever, and Gramps forever lonely.
*   [ ] Design a rematch team for Eusine after you've captured Ho-Oh (or after capturing all three beasts). Or maybe add a way to have unlimited rematches, like the Saffron Dojo. (e.g. Eusine goes to Bellchime Trail on Wednesday nights)
*   [ ] Testing

### CODE REVIEW

File: maps/EusinesHouse.asm [hunk 1]
```diff
@@ -183,3 +187,14 @@ EusinesHouseGrampsText2:
...
 	cont "ary #mon!"
 	done
+
+EusinesHouseGrampsText3:
+	text "My grandson Eusine"
+	line "is on another"
+	cont "adventure."
+
+	para "He left to visit"
+	para "Ecruteak city"
```
Comment (Rangi42) [inline maps/EusinesHouse.asm:197]:
```suggestion
	para "Ecruteak City"
```

**Comment by Rangi42** [comment]
re: todo, I'll design a rematch team for him after you've captured Ho-Oh (or maybe just captured all three beasts).

**Comment by Rangi42** [comment]
> I think there should be two rematches. One after capturing Suicune, and another after capturing Ho-Oh.

I'd rather not; I think it would be too easy for players to miss one or the other. Maybe I'll add a way to have unlimited rematches, like the Saffron Dojo. (e.g. Eusine goes to Bellchime Trail on Wednesday nights)

File: maps/EusinesHouse.asm [hunk 2]
```diff
@@ -93,8 +93,12 @@ CeladonEusine:
...
 EusinesHouseGrampsScript:
-	checkevent EVENT_SET_WHEN_FOUGHT_HO_OH
-	iftrue_jumptextfaceplayer EusinesHouseGrampsText2
+	checkevent EVENT_EUSINES_HOUSE_EUSINE
+	iffalse, .ishome
+	checkevent EVENT_FOUGHT_HO_OH
+	iftrue_jumptextfaceplayer EusinesHouseGrampsText3
+	jumptextfaceplayer EusinesHouseGrampsText2
+.ishome:
 	jumptextfaceplayer EusinesHouseGrampsText1
```
Comment (Rangi42) [inline maps/EusinesHouse.asm:102]:
```suggestion
	checkevent EVENT_EUSINES_HOUSE_EUSINE
	iffalse_jumptextfaceplayer EusinesHouseGrampsText1
	checkevent EVENT_FOUGHT_HO_OH
	iftrue_jumptextfaceplayer EusinesHouseGrampsText3
	jumptextfaceplayer EusinesHouseGrampsText2
```

File: maps/EusinesHouse.asm [hunk 3]
```diff
@@ -183,3 +187,14 @@ EusinesHouseGrampsText2:
 	line "about the legend-"
 	cont "ary #mon!"
 	done
+
+EusinesHouseGrampsText3:
+	text "My grandson Eusine"
+	line "is on another"
+	cont "adventure."
+
+	para "He left to visit"
```
Comment (Rangi42) [inline maps/EusinesHouse.asm:196]: This seems to imply you'd see him in Ecruteak, but you shouldn't.

Comment (vulcandth) [reply maps/EusinesHouse.asm:196]: Should it be more specific and say that he left to visit the Tin Tower in Ecrutreak City?

Comment (Rangi42) [reply maps/EusinesHouse.asm:196]: If he's in Bell Tower at that time, yes.

**Pull request closed without merging**